### PR TITLE
[5.2] Register console application via container

### PR DESCRIPTION
**Pull request by themsaid**
This PR registers and resolves `Illuminate\Console\Application` via the container.

Not sure if `Illuminate\Foundation\Providers\ArtisanServiceProvider` is the right place for this though.

### CODE REVIEW

File: src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php [hunk 1]
```diff
@@ -106,6 +106,10 @@ class ArtisanServiceProvider extends ServiceProvider
      */
     public function register()
     {
+        $this->app->singleton('artisan', function ($app) {
+            return new \Illuminate\Console\Application($app, $app['events'], $app->version());
+        });
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php]: Please import this.

**Comment by GrahamCampbell** [comment]
This change is breaking for those that were extending getArtisan in order to use a custom console application class. Please send to 5.3.

**Comment by themsaid** [comment]
@GrahamCampbell can you explain how this is breaking? I only resolved the instance from the container instead of creating it inside the method. Anyone extending `getArtisan()` can still use the extension.

**Comment by GrahamCampbell** [comment]
But now, you've bound `'artisan'` to the container, and it might not even be the correct instance.

**Comment by themsaid** [comment]
@GrahamCampbell I see, your are right. Thanks for the clarification.

**Pull request closed without merging**